src/app/modules/OpenshiftStreams/components/StreamsTableConnected/useKafkaInstances.ts

```diff
@@ -196,3 +196,16 @@ function valuesToQuery(
     )
     .join(" or ");
 }
+
+function apiStatusToUIStatus(status: string): Status {
+  const mapping: { [key: string]: Status } = {
+    accepted: "accepted",
+    preparing: "preparing",
+    provisioning: "provisioning",
+    ready: "ready",
+    failed: "degraded",
```

Comment:
Why failed is mapped as `degraded`?

Reply:
> 'm failing (pun intended) 
 
 ha ha :-)

>  I think failed has been considered too strong in its wording. I'm failing (pun intended) at finding the paper trail behind this choice, as usual @jgiardino might know more

Thanks. It is one of those statuses that have a double meaning:
- the one listed here of an existing instance being degraded and it'll become ready/available again 
- and the other meaning is that a new instance deployment has actually failed and there is no possibility of that kafka being ready. 

This is being discussed more on https://issues.redhat.com/browse/MGDSTRM-7307 and I was curious to see if this was ever brought up in that epic brief.

Reply:
Adding @pb82 to this one. He kicked off some discussion via email with the subject thread "[mas-devel] Unstable Kafka instances and how a customer would see them". There is some discussion in that thread about how the failed status doesn't accurately communicate that state. For example:

> does it ever make sense to move an instance from ready to failed (from a UX perspective)?
> In normal operation of the system, this should never happen. .  The instance once ready for the first time should always be offering at least two brokers until it is deleted.  In cases it does happen currently, there will be a defect at root cause (or some kind of catastrophic infrastructure failure (loss of 2 AZ, loss of connection to storage).

> I think what's potentially confusing about the "Failed" state is that it sounds terminal - it's not going to come back from this - but that actually it's often transient (if it's non-creation error). I think differentiating between permanent errors (Failure) and transient errors (Currently unavailable).

> To the best of my knowledge, we actually don't have any terminal failure states at present.  There's no way for a user to cause fleet-manager to generate a request that can't be processed and the fleet manager should only accept requests for which there is capacity.  At the moment I think all failed conditions should be classified as Currently Unavailable. 

Additional discussion happened in this jira, where Katie kicked off a proposal for what to call this in the first comment:
https://issues.redhat.com/browse/MGDSTRM-7231

We then moved forward with creating UI components that used this label, but I don't think a jira was ever opened to update this label across our documentation or in the API.

@pb82 Going back to your original thread on this, I assume your question was to determine if the label in the API should change. What are your thoughts on using "Degraded"? Are there any other action items that you think need to happen before making this switch across the UI, API, and documentation?

Reply:
Also, @machi1990 - Thanks for sharing https://issues.redhat.com/browse/MGDSTRM-7307
Reading through the comments, it isn't clear if any decisions have been made on what changes to status we might have. Do you know if any of the suggestions there are finalized, or is it still up for discussion?

Reply:
I think the discussion are nearly finalized and there is a path forward, it is just that the work is not yet progressed implementation wise. I'll defer to @pb82 for further feedback has he has been involved in the discussions.

Reply:
Thanks! If either of you have a link to what the proposed status updates are along with what they mean (how does this status happen, is the user blocked or will it resolve itself), please share them. 

My initial thoughts are that the following status from that jira (7307) would use the same visual styling that we have now for "Degraded" with the warning icon:
> "Ready - limited capacity" - when kafka instance is available but not all brokers are running

And for this one, I'm not sure if it should also be a warning or error icon. I think that would depend on if it can become available/ready again.
> "Unavailable" - when kafka instance is not usable

It's also worth noting that David Ffrench had some comments in [MGDSTRM-7231](https://issues.redhat.com/browse/MGDSTRM-7231) about these types of statuses and providing some messaging to the user about what this means (are we working to resolve the situation? does the user need to do anything?)